framework/include/base/MooseError.h

```diff
@@ -240,27 +240,37 @@ mooseDeprecatedStream(S & oss, const bool expired, const bool print_title, Args
 
   std::ostringstream ss;
   mooseStreamAll(ss, args...);
-
-  std::string msg =
-      print_title
-          ? mooseMsgFmt(
-                ss.str(),
-                "*** Warning, This code is deprecated and will be removed in future versions:",
-                expired ? COLOR_RED : COLOR_YELLOW)
-          : mooseMsgFmt(ss.str(), expired ? COLOR_RED : COLOR_YELLOW);
-  oss << msg;
-  ss.str("");
-  if (Moose::show_trace)
+  size_t pos = ss.str().find('\n');
+  if (pos != std::string::npos)
   {
-    if (libMesh::global_n_processors() == 1)
-      print_trace(ss);
-    else
-      libMesh::write_traceout();
+    auto hash = std::hash<std::string>{}(ss.str().substr(pos + 1));
+    auto iter = Moose::_deprec_tracker.find(hash);
```

Comment:
system by system is the way to go, not a global attribute 

This is also relying on convoluted logic
"hash the part of the message before the line break" that works fine now but could be a pain to debug when it stops working

This commit should be reverted. What kind of issue are you having with the factory tracking?

Reply:
There is no technical issue with implementing the object tracking in the factory system per se. 

My issue is more that we have to create two different systems in two different places that don't really interact with each other in order to keep track of their deprecation outputs. Instead of doing that, I figured it would be better to have all of them stored in one singular location that doesn't differentiate between what kind of deprecated output it is, just that `mooseDeprecatedStream` was called. Doing it this means that if we add new kinds of deprecation output, the system would still work with no modifications needed, and a universal tracker leaves a smaller footprint.

Design-wise, this does seem like a better solution. If the primary issue with the universal tracker is the line break check being too brittle, then that should be fixable.

Reply:
it s not a lot of code to duplicate here. Placing code in Moose.C is an easy solution but it s not sustainable. We need to distribute the codes between systems or concentrate it in utils if there's good redundancy.

in general we should shun global variables. There's threading issues you need to be wary of when you use global variables, it s not worth it

Reply:
@loganharbour

Reply:
Let s have a look. Where is this triggered for creating objects?